src/Host/Client/Impl/Host/LocalBrokerClient.cs

```diff
@@ -24,14 +24,15 @@ internal sealed class LocalBrokerClient : BrokerClient {
         private const string InterpreterId = "local";
 
         private static readonly bool ShowConsole;
-        private static readonly LocalCredentialsDecorator Credentials = new LocalCredentialsDecorator();
+        private static readonly LocalCredentialsDecorator _credentials = new LocalCredentialsDecorator();
 
         private readonly string _rhostDirectory;
         private readonly string _rHome;
         private readonly BinaryAsyncLock _connectLock = new BinaryAsyncLock();
         private readonly ICoreServices _services;
 
         private Process _brokerProcess;
+        private int _disposed;
```

Comment:
It was throwing for me in some cases when object was disposed by the time `Add` was called.

---
In reply to: [82297812](https://github.com/Microsoft/RTVS/pull/2447#discussion_r82297812) [](ancestors = 82297812)